src/main/fc/config.c

```diff
@@ -1297,6 +1289,34 @@ void validateAndFixGyroConfig(void)
         if(motorConfig()->motorPwmRate > maxEscRate)
             motorConfig()->motorPwmRate = maxEscRate;
     }
+
+    uint32_t gyroFrequencyNyquist = (1.0f / (samplingTime * gyroConfig()->gyro_sync_denom)) / 2;
+
+     if (gyroConfig()->gyro_soft_lpf_hz > gyroFrequencyNyquist) {
+         gyroConfig()->gyro_soft_lpf_hz = 0;
```

Comment:
I think we're getting to the core of the problem. If, instead of entirely disabling the filters, we just set them to the Nyquist frequency, then @borisbstyle's solution sounds quite acceptable from a user perspective to me.
This way, if a user has a craft that needs filtering to be flyable, it will not just become unflyable when the loop time is increased.